### DISCUSSION TITLE
Dịch `stratified sampling`

### DISCUSSION THREAD

**Discussion by cuongvng** [mlbvn/handson-ml2-vn#12] [Terminology] [answered]
Từ này mình không biết nên dịch như nào hay để nguyên. Xin ý kiến mọi người.

cc: @mlbvn/handson-ml2

**Comment by lkhphuc** [comment]
`lấy mẫu theo loại` được không?

**Comment by minhduc0711** [comment]
mình search thì bên kinh tế họ hay dịch là `lấy mẫu phân tầng` thì phải, nhưng mà từ `tầng` nghe có vẻ không hợp lý lắm.

**Reply by ngcthuong** [reply]
Lấy mẫu nhóm? À không phải, vì nó còn phân biệt với từ "cluster sampling"

**Reply by ngcthuong** [reply]
Stratified sampling có điều kiện là mỗi nhóm đều lấy cùng một số lượng chứ không lấy theo population của từng nhóm, và các nhóm không chồng lấn nhau. 

Lấy mẫu tách nhóm?

**Reply by hnguyentt** [reply]
Em thấy `lấy mẫu phân tầng` hợp lý hơn và cũng gần nghĩa góc của tiếng Anh. Nếu gọi là `lấy mẫu tách nhóm` thì em sẽ nghĩ là tách ra từng nhóm rồi chọn một vài nhóm trong các nhóm đó như `cluster sampling`.

`stratified sampling` là mình chia dữ liệu ra nhiều phần (trong tiếng anh họ dùng thuật ngữ là `stratum`, aka `tầng` trong tiếng Việt), việc tách tầng này có thể là theo `loại` hoặc **theo thêm những dữ kiện khác**. Ví dụ tách ảnh X-ray để dự đoán ảnh bị các loại bệnh A, B hay không. Người ta có thể tách theo các tầng (theo loại) sau đó chọn tách từng tầng này thành 3 phần cho `train`, `val`, `test`.
* <a href="https://www.codecogs.com/eqnedit.php?latex=\rm&space;A,&space;\overline&space;{\rm&space;B}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\rm&space;A,&space;\overline&space;{\rm&space;B}" title="\rm A, \overline {\rm B}" /></a>
* <a href="https://www.codecogs.com/eqnedit.php?latex=\overline&space;{\rm&space;A},&space;B" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\overline&space;{\rm&space;A},\rm&space;B" title="\overline {\rm A}, \rm B" /></a>
* <a href="https://www.codecogs.com/eqnedit.php?latex=\rm&space;A,&space;\rm&space;B" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\rm&space;A,&space;\rm&space;B" title="\rm A, \rm B" /></a> 
* <a href="https://www.codecogs.com/eqnedit.php?latex=\overline{\rm&space;A},&space;\overline{\rm&space;B}" target="_blank"><img src="https://latex.codecogs.com/gif.latex?\overline{\rm&space;A},&space;\overline{\rm&space;B}" title="\overline{\rm A}, \overline{\rm B}" /></a>

Trong trường hợp này như đề xuất của anh @lkhphuc `lấy mẫu theo loại` thì hợp lý. Nhưng người ta cũng có thể tách tầng theo cả những đữ liệu về tuổi tác, giới tính, nghề nghiệp, dân tộc, etc. của bệnh nhân sao cho distribution của tất cả các yếu tố này là như nhau trong 3 tập train, val, test. Lúc này mình sẽ chia nhiều `stratum` hơn để lấy mẫu. Nếu dùng `lấy mẫu theo loại` thì người đọc sẽ hiểu theo nghĩa hẹp hơn.

**Reply by cuongvng** [reply]
em nghĩ dùng `tầng` thì dễ bị hình dung ra tính thứ bậc cao thấp (kiểu hierarchical), trong khi các stratum về cơ bản là độc lập và bình đẳng. Em thấy `lấy mẫu tách nhóm` hoặc `lấy mẫu phân nhóm` ổn, `cluster` thì đã thống nhất dịch là `cụm` rồi nên em nghĩ không có vấn đề gì.

**Reply by forummlcb** [reply]
Mình chưa thấy lựa chọn nào hợp lý lắm, nếu giữ nguyên là "lấy mẫu stratified" thì sao nhỉ? Thêm dòng chú thích nữa.

Mình thấy nếu một từ khó thống nhất thì cứ giữ nguyên tiếng Anh rồi sau này search và đổi sau. Như thế dễ hơn là chuyển qua tiếng Việt vì search sẽ khó hơn.

**Reply by ngcthuong** [reply]
Ý ở đây là cách lấy mẫu khi phân các nhóm / tầng là lấy mẫu như nhau. Còn nếu gọi là tầng thì nó vẫn phân theo kiểu phân bố hơn như @cuongvng  nói. Cái cần là cần nổi ý là các nhóm được phân ra là không chồng lấn. 

Mình nghĩ tạm thời cứ để lại rồi mình tính sau.

**Reply by forummlcb** [reply]
Theo mình thì trong cuốn này những từ nào khó ra quyết định thì cứ giữ nguyên tiếng Anh để không ảnh hưởng tới tiến độ.

**Reply by cuongvng** [reply]
vậy tạm thời em sẽ giữ nguyên.

**Comment by thanhcsf** [comment, marked as answer]
Tạm thời giữ nguyên.
